mathesar/tests/views/api/test_table_api.py

```diff
@@ -1264,3 +1268,136 @@ def test_table_patch_columns_invalid_type_with_multiple_changes(create_data_type
     # The table should not have changed
     original_column_data = _get_data_types_column_data()
     _check_columns(current_table_response.json()['columns'], original_column_data)
+
+
+def test_table_delete_no_primary_key(create_schema, client):
+    table_name = 'NASA Table Delete'
+    schema = create_schema('Patents')
+    engine = create_mathesar_engine(schema.database.name)
+    metadata = MetaData(bind=engine, schema=schema)
+    columns = [
+        Column('id', Integer),
+        Column('name', String)
+    ]
+    table = Table(
```

Comment:
This needs to be a SQLAlchemy table, not `mathesar.models.Table`. You'll need to import it from SQLAlchemy (to avoid a name conflict, you should do something like `from sqlalchemy import Table as SATable)

Then you would change this line to `sa_table = SATable(`